Ddp 8511 update medical provider table

gskhirel opened the pull request:
https://broadinstitute.atlassian.net/jira/software/c/projects/DDP/boards/836?modal=detail&selectedIssue=DDP-8511
1.After autocompletion for physician component is done, Patch request for medical provider should be updated
2. Updated Medical provider Dtos and table by adding country field

gskhirel committed:
[DDP-8511] update medical provider request

gskhirel committed:
revert line endings

ssettipalli commented on pepper-apis/dss-database/src/main/resources/db-changes/schema/DDP-8511-physician-country-add.xml:
```diff
@@ -0,0 +1,14 @@
+
+    <changeSet author="gskhirel" id="physician_country_add">
+        <addColumn tableName="user_medical_provider">
+            <column name="country" type="varchar(100)">
```
wondering if it should be country or country_id (reference to country) like we have in mailing_address .
I think country should be fine though. Everything else looks to me.
@MocanaAtBroad : any thoughts on this ?

MocanaAtBroad replied:
I would encourage you to use country_address_info table. Has all the countries in the world and we have translations for them. You can also do autocompletion.
![Screen Shot 2022-08-16 at 8 51 26 AM](https://user-images.githubusercontent.com/29984380/184884071-657463c7-f7d2-458e-afdf-807a614db4da.png)

gskhirel committed:
Merge branch 'develop' into DDP-8511-update-medical-provider-table

ssettipalli commented:
Most important thing to watchout is:
This component is widely used by ALL studies. 
We might not be deploying angular/front-end for ALL existing studies and changes should be backward compatible and not break any existing studies if front-end/angular is not updated.

bskinner commented on pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/route/PatchMedicalProviderRoute.java:
```diff
@@ -66,6 +66,7 @@ public Object handle(
                     String physicianName;
                     String city;
                     String state;
+                    String country;
                     try {
                         JsonElement data = new JsonParser().parse(request.body());
                         if (!data.isJsonObject() || data.getAsJsonObject().entrySet().size() == 0) {
```
Any API changes must also have their associated documentation updated in `pepper-apis/docs/specification`.

gskhirel committed:
[DDP-8511] change country column to id

gskhirel committed:
Merge branch 'develop' into DDP-8511-update-medical-provider-table

gskhirel committed:
change column name

gskhirel committed:
Merge remote-tracking branch 'origin/DDP-8511-update-medical-provider-table' into DDP-8511-update-medical-provider-table

gskhirel committed:
Merge branch 'develop' into DDP-8511-update-medical-provider-table

bskinner commented on pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/util/DataLoader.java:
```diff
@@ -732,6 +738,10 @@ String getMedicalProviderGuid(Handle handle) {
                 MedicalProviderTable.TABLE_NAME, MedicalProviderTable.MEDICAL_PROVIDER_GUID);
     }
 
+    long getCountryAdressId(Handle handle, String countryName) {
```
There's a typo here- this should be `Address`, not `Adress`

bskinner commented on pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/util/DataLoader.java:
```diff
@@ -688,6 +692,7 @@ public void loadReleaseSurveyData(Handle handle, ParticipantData participantData
                 || releaseSurvey.getInitialBiopsyCity() != null
                 || releaseSurvey.getInitialBiopsyState() != null) {
             String guid = getMedicalProviderGuid(handle);
+            long countryAdressId = getCountryAdressId(handle, releaseSurvey.getCountry());
```
Should be `Address`, not `Adress`

bskinner commented on pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/db/dao/JdbiMedicalProvider.java:
```diff
@@ -211,6 +217,9 @@ List<MedicalProviderDto> getAllByUserGuidStudyGuid(
     );
 
+    @SqlQuery("SELECT country_address_info_id from country_address_info where name =: name")
+    long getCountryAddressInfoId(@Bind("name")String countryName);
```
There should be a space between `("name")` and `String`

bskinner commented on pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/util/DataLoader.java:
```diff
@@ -666,6 +668,7 @@ public void loadReleaseSurveyData(Handle handle, ParticipantData participantData
             for (Institution institution : institutionList) {
 
                 String guid = getMedicalProviderGuid(handle);
+                long countryAdressId = getCountryAdressId(handle, institution.getCountry());
```
The `attach()` call made by `getCountryAddressId()` is expensive- try to avoid calling it multiple times withing the scope of a single method.

bskinner commented on pepper-apis/dss-database/src/main/resources/db-changes/schema/DDP-8511-physician-country-add.xml:
```diff
@@ -0,0 +1,18 @@
+
+    <changeSet author="gskhirel" id="physician_country_add">
+        <addColumn tableName="user_medical_provider">
+            <column name="country_address_info_id" type="bigint" >
+            </column>
```
Let's use `<column .../>` instead of `<column...></column>` here

bskinner commented on pepper-apis/dss-core/src/test/java/org/broadinstitute/ddp/route/PostMedicalProviderRouteTest.java:
```diff
@@ -267,7 +271,7 @@ private static final class TestMedicalProviderData {
         public static final String CITY = "West Windsor Township";
         public static final String STATE = "New Jersey";
-
+        public static final String COUNTRY = "United States";
```
Using the 2 letter codes would be preferred to using the readable name in the request payload. Our 2-letter country code columns try to adhere to the `ISO 3166-1 alpha-2` standard.

gskhirel replied:
After autocompletion, suggested countries are fully written, populated data of country names in country_address_info table is also written as a full country name. When post request is being sent country names are written fully and we do search for corresponding ids in  country_address_info via full name, therefore if we use ISO 3166-1 alpha-2 standard, we might mess up this test.

bskinner replied:
Where are the autocompletion details coming from?

gskhirel replied:
It comes from : institution, city, country_subnational_division, country_address_info tables.

gskhirel committed:
Merge branch 'develop' into DDP-8511-update-medical-provider-table

gskhirel committed:
[DDP-8511] update loader

gskhirel committed:
Merge remote-tracking branch 'origin/DDP-8511-update-medical-provider-table' into DDP-8511-update-medical-provider-table

gskhirel committed:
Update DataLoader.java

gmakharat committed:
[DSM] DDP-7477 | use naive approach to add first name, last name, dob (#2166)

* DDP-7477 | use naive approach to add first name, last name and date of birth for sample queues

* DDP-7477 | clean code a little bit

Co-authored-by: Guram Makharashvili <gmakhara@Gurams-MacBook-Pro.local>

dgiorbel committed:
fixed reconsent bug for aged up users and for os1 users (#2173)

Trimax committed:
DDP-8578. Order participants (#2172)

Co-authored-by: Dmitrii Eliseev <deliseev@broadinstitute.org>
Co-authored-by: Alex Maratkanov <alextanlan@gmail.com>

gskhirel committed:
change physician dataloader data to country

devonbush committed:
DDP-7866: fixing edge case around text question handling (DSM) (#2174)

DDP-7866: fixing edge case around text question handling

bskinner commented on pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/model/migration/Physician.java:
```diff
@@ -20,6 +20,9 @@ public final class Physician {
     @SerializedName("state")
     private String state;
 
+    @SerializedName("country")
+    private String country;
+
```
What is the expected behavior if a client passes `null` for this field? Given there are a number of clients using this API, and they may not all be updated, this should be backwards compatible with any currently deployed sites.

gskhirel replied:
Actually we discussed the scenario, when an user manually enters the country name which is not present in country_address_info, if we consider the current state of user_medical_provider table that kind of data will be lost. So maybe it will be better if we revert the column type back to varchar and put country names directly without relations?

gskhirel replied:
This issue is also described in https://broadinstitute.atlassian.net/browse/DDP-8631

bskinner replied:
I talked this over with @MocanaAtBroad- let's revert this back to just storing the bare string in the database, and not worrying about making sure it's a proper foreign reference.

LKapanad committed:
DDP-8586 osteo ageup addendum (#2169)

* DDP-8586 add intermediate activity for ageup consent-addendum event

* DDP-8586 fix ageup events

* DDP-8586 fix osteo family-history

* DDP-8586 fix medical release precondition after addendum

* DDP-8586 refactor about child activity

* DDP-8586 fix templates

* DDP-8586 fix family history creation events

* DDP-8586 fix family history patch

* DDP-8586 fix family history events

* DDP-8586 remove child family history activity

* DDP-8586 fix subs.conf

ssettipalli committed:
DDP-8616. update email template text (#2177)

devonbush committed:
DDP-8628: adding explicit 'human readable' export format (#2176)

ssettipalli committed:
DDP-8218. remove pancan followup activity creation events. (#2165)

pegahtah committed:
[DSM] DDP-7556 Mercury Sequencing Orders (#1873)

* DDP-7556
* added the PR comments
* made the request to work with collaborator pt id as well

* DDP-7556
* fixed the checkstyle error

* DDP-7556
* moved try catch into `createAndPublishMessage` method and removed it from Dao

* DDP-7556
* moved the `createAllOrders` before the pubsub publishing to check for existing

* DDP-7556
* added the ability to always have test tissues with sample id

* Applying PR comments and fixes to the branch
*added mercuryOrderCreator to DDPInstanceDao
*fixed the SQL query for getting eligible tissues and samples
*fixed the json to have all the info in a `productOrder` object
*changed published json on pubsub to be base64
*added userId and ddpInstanceId columns to the backend to keep better track of the orders

* Applying PR comments and fixes to the branch
*fixed merge conflicts

* Applying PR comments and fixes to the branch
*renamed Route to`PostMercuryOrderDummyRoute`
* removed `ddpParticipant Id` from `MercuryOrderDummyRequest`
* renamed `MercuryOrderRequest` class to `MercuryOrderDummyRequest`

* I added a Dao class `MercurySampleDao.java` for getting eligible samples for ordering per participant, and also the `GetMercuryEligibleSamplesRoute.java` route
 I added the instance and permission roles needed
 Also created the MercurySampleDto

* Adding collectionDate

* Adding collectionDate
Fixing bug in search page to bring back all of the kits

* Added `PostMercuryOrderRoute` to place a sequencing order from frontend, also added methods to `TissueSMIDDao.java` and `KitRequestDao.java`
Added checks for getting only tissues that don't have a missing sm id in uss and scrolls in `MercurySampleDao.java`
Added `createBarcodes` function in `MercuryOrderUseCase.java` to get all the sm ids and the kit label
Added `MercuryOrderStatusListener` for listening to pubsub subscription

* * added ClinicalOrderDao for getting the Clinical Orders per realm for the Clinical Order's page
* added two columns of tissue_id and dsm_kit_Request_id instead of sample_id and changed the Dao and Dto classes accordingly
* Corrected bug in MercurySampleDao.java to avoid getting tissues more than once
* Added the ClinicalOrderUseCase.java to publish an ActionRequestMessage and also added the ActionRequestMessage
* I removed the base64 Encoding by Gp's request
* Added the functionality of getting orders in the GetMercuryEligibleSamplesRoute.java
* Added checks for access in all the routes
* Added the instance role in the DisplaySettingsRoute
* Added a changeset of `mercury_sequencing_further_changes` for changes to the table and foreign keys

* * Adding the MercuryOrderStatusListener to DSMServer

* adding `sequencing_restriction` field

* removing the FON test route

* adding sequencingRestriction to the object

* added test for accessioning
added mappings for new fields

* added tracking times and checks for accessioning
added mappings for new fields

* DDP-8217 adding the new columns in sm id table and updating them
DDP-8217 adding date_px ad collectionDate for sm ids
DDP-7556 DisplaySettingsRoute returns if the realm has sequencting

* DDP-8217 only adding sm ids with received_date (Accessioned) to the order

* DDP-7604 Since collection date field is added, we have to make sure dummy kits also have that

* DDP-7556
PR Comments

* DDP-7556
PR Comments

* DDP-7556
PR Comments

* DDP-7556 PR comments
* Changing the return type from `ArrayList` to `List` in a number of classes
* adding `@ColumnName(DBConstants.COLLECTION_DATE)` for CollectionDate in `KitRequestShipping`
* removing static from `getInstanceRoles` method
* removing typo in name `processOrderStatus()`
* moving `clinicalOrderUseCase.publishStatusActionMessage(orders, projectId, topicId);` to GetMercuryOrdersRoute.processRequest
* adding  check for Failed messages
* changing the thrown exception in `updateOrderStatus` to `DSMPubSubException`
* removing unnecessary `static` in `MercuryOrderPublisher`

* DDP-7556
* removing unused fields

* merging develop and removing old `DownloadParticipantFileRoute`

* DDP-7556 not passing connection to `MercuryOrderDao.create` method anymore
adding the `BY_INSTANCE_NAME` query extension to `SQL_SELECT_ROLES_FOR_INSTANCE` in `DDPInstanceDao`

* DDP-7556 updated mapping and created migrator

* DDP-7556 added filtering ability for ClinicalOrders of a particiapnt

* DDP-7556 name change in mapping to `clinicalOrder`

* DDP-7556
Adding the functionality to export to ES a new published order
Changing type of `orderId` to keyword

* DDP-7556 checkstyle fix

* DDP-7556
Exporting Status updates

* DDP-7556
fixed a bug in query

* DDP-7556
Exporting the two new fields

* DDP-7556
Exporting Status updates

bskinner commented on pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/route/PatchMedicalProviderRoute.java:
```diff
@@ -107,6 +107,7 @@ public Object handle(
                             physicianName,
                             city,
                             state,
+                            existingMedicalProviderDto.getCountryId(),
```
This is following a different pattern from the rest of the parameters included with the payload. Is there a particular reason for the deviation?
